olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -400,7 +414,12 @@ HttpResponse OlpClient::CallApi(
   auto response =
       SendRequest(network_request, settings_, retry_settings, context);
 
-  for (int i = 0; i < retry_settings.max_attempts && !context.IsCancelled();
+  // Make sure that we don't wait longer than `timeout` in retry settings
+  long long accumulated_wait_time = backdown_period;
+  const long long max_wait_time =
+      retry_settings.timeout * 1000ll;  // convert seconds to msecs
```

Comment:
```suggestion
      retry_settings.timeout * 1000ll;
```
Can be removed if a self-explanatory abstract type used.